### PR TITLE
LPS-155455, LPS-155459: Create automated tests for FDS

### PR DESCRIPTION
**Pull request by mateusralv**
**Background**
Create automated tests CanDisplayOnHover and CanDisplayOnMultipleActiveRows

**Test Plan**
FDSQuickActions#CanDisplayOnHover ([LPS-155455](https://issues.liferay.com/browse/LPS-155455))
Given: frontend-data-set-sample-web (FDS Sample Portlet) deployed
And Given FDS Sample Portlet added to page
When: Hover over 1st row item (Sample 1)
Then: quick action menu is displayed in the 1st row

FDSQuickActions#CanDisplayOnMultipleActiveRows - [LPS-155459](https://issues.liferay.com/browse/LPS-155459)
Given: frontend-data-set-sample-web (FDS Sample Portlet) deployed
And Given: FDS Sample Portlet added to page
When click on ellipsis button 2nd row item
Then the quick action menu is displayed on the 2nd row
When hovering over the 1st row item
Then the quick action menu is displayed on the 1st row
And Then the quick action menu is not displayed on the 2nd row

**JIRA Ticket:**
[LPS-155455](https://issues.liferay.com/browse/LPS-155455)
[LPS-155459](https://issues.liferay.com/browse/LPS-155459)

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 1]
```diff
@@ -10,6 +10,31 @@
...
+</tr>
+<tr>
+	<td>DROPDOWN_MENU</td>
+	<td>//div[@class='dropdown-menu show']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>VERTICAL_ELLIPSIS_BUTTON</td>
+	<td>//div[(text()='${key_title}')]//following-sibling::div//button[contains(@class,'dropdown-toggle')]</td>
+	<td></td>
```
Comment (susanchen3) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:36]: this path already exists :eyes:

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
Looks good to me! :sunglasses: Will need to rebase after https://github.com/brianchandotcom/liferay-portal/pull/119280 merges

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
Will want to have some evidence that the tests pass on CI. Either have passing ci run on the PR or attach a poshi report of local passing test to JIRA ticket.

**Comment by mateusralv** [comment]
Ok, thanks John, I'm going to remove the button that's duplicated, and then I'll add the index file to the ticket. 😃

**Comment by john-co** [comment]
Hi @mateusralv , do you have the poshi report files?

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
> Hi @mateusralv , do you have the poshi report files?

Hi @john-co, I updated the tickets ([LPS-155459](https://issues.liferay.com/browse/LPS-155459) and [LPS-155455](https://issues.liferay.com/browse/LPS-155455)) and put poshi report files.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase [hunk 2]
```diff
@@ -65,6 +65,42 @@ definition {
...
+	@priority = "3"
+	test CanDisplayOnMultipleActiveRows {
+		task ("When click on Elipsis button 1st") {
+			Click(
+				key_title = "Sample1",
+				locator1 = "FrontendDataSet#VERTICAL_ELLIPSIS_BUTTON");
+		}
+
+		task ("Then the quick action menu is displayed on the 1st line") {
+			AssertElementPresent(locator1 = "FrontendDataSet#DROPDOWN_MENU");
```
[the remark this answers is not in the document]
Comment (john-co) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase:86]: Why not use `FDSQuickActions.viewQuickActionMenuIsPresent(title = "Sample1");` like you did for Sample2?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase [hunk 3]
```diff
@@ -65,6 +65,42 @@ definition {
...
+				key_itemName = "Sample2",
+				locator1 = "FrontendDataSet#TABLE_ITEM_ROW");
+		}
+
+		task ("Then the quick action menu is displayed on the 2nd line") {
+			FDSQuickActions.viewQuickActionMenuIsPresent(title = "Sample2");
+		}
+
+		task ("Then the quick action menu is displayed on the 1st line") {
+			AssertElementPresent(locator1 = "FrontendDataSet#DROPDOWN_MENU");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase:100]: same comment

**Review by john-co — CHANGES_REQUESTED** [review]
Hi @mateusralv, please review comments.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase (hunk 3 above)
Comment (mateusralv) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSQuickActions.testcase:100]: @john-co I thought was to confirm that both menus (blue and red) were being shown. That's why I used the dropdown and not the `FDSQuickActions.viewQuickActionMenuIsPresent `macro. So I confused and would only need to confirm that the quick menu (blue) is visible?
FDS quick menu

**Review by john-co — CHANGES_REQUESTED** [review]

**Comment by john-co** [comment]
> @john-co It's not possible, for me, to show that quick actions on two rows at the same time, would it be a bug? thinking So I thought it would be the quick menu and the dropdown menu.
> 
>  FDS.Quick.menu.mp4

Based on the message [here](https://github.com/liferay-frontend/liferay-portal/pull/2262#issuecomment-1144794546), 

> Extraball, the user could leave the menu opened and move over other rows. But, of course, these rows must show the hover actions too.

I'm expecting that the quick actions would remain on the row with open menu, and another quick actions on the other row that is moused over. Can you confirm what we should expect @drakonux ?

**Comment by john-co** [comment]
> @john-co It's not possible, for me, to show that quick actions on two rows at the same time, would it be a bug? thinking So I thought it would be the quick menu and the dropdown menu.
> 
>  FDS.Quick.menu.mp4

Hi @mateusralv , while we're waiting for design decision I'll update the test to current behavior so we have something in CI for now.

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
Thanks @drakonux , created a bug ticket for it [here](https://issues.liferay.com/browse/LPS-157838)

**Pull request closed without merging**